src/internal/middleware/auth/handler.go

```diff
@@ -75,3 +78,12 @@ func GetWhoAmI(ctx context.Context) string {
 func setWhoAmI(ctx context.Context, username string) context.Context {
 	return context.WithValue(ctx, whoAmIResultKey, username)
 }
+
+// AsInternalUser gives a context's a cached whoami username of form internal:<name>. It also
+// strips away existing incoming metadata to add an empty auth token. As a result, this context will
+// not be able to make additional gRPCs
+func AsInternalUser(ctx context.Context, username string) context.Context {
+	emptyToken := metadata.New(map[string]string{auth.ContextTokenKey: ""})
```

Comment:
Hopefully this doesn't create any glaring issues

Reply:
I think based on the implementation of grpc metadata we could just do 
```
emptyToken := metadata.New(map[string]string{})
```

to have zero grpc metadata. We could also reset the outgoing context, just to make sure (if the user uses client.Ctx() it'll have outgoing auth metadata copied from the incoming request).

Reply:
I specifically added that key because `getAuthenticatedUser` checks that before the cached value and errors if it's not present. Could also switch those checks but I thought there might be a good reason. Clearing outgoing metadata makes sense to me. Agreed about comment - do you think it could make sense to add logic that prevents "internal:" users from being granted permissions? That would limit the damage it could do (though would complicate things if spec repos go back to requiring a role).

Reply:
We can double-check with @acohen4 but checking the cached value before the grpc metadata makes sense to me. 

Long-term I'd like to be able to grant internal users permissions. I actually don't think it's possible for a user to modify an internal user's permissions today because we validate that the users in a role binding have a valid prefix.

Reply:
makes sense to me too 👍